packages/next/src/server/dev/hot-reloader.ts

```diff
@@ -413,6 +471,31 @@ export default class HotReloader {
       )
   }
 
+  private async getVersionInfo(span: Span) {
+    const versionInfoSpan = span.traceChild('get-version-info')
+    return versionInfoSpan.traceAsyncFn<VersionInfo>(async () => {
+      try {
+        const installed = require('next/package.json').version
+
+        const res = await fetch(
+          'https://registry.npmjs.org/-/package/next/dist-tags'
+        )
+
+        if (!res.ok) return { installed, staleness: 'unknown' }
+
+        const tags = await res.json()
+
+        return parseVersionInfo({
+          installed,
+          latest: tags.latest,
+          canary: tags.canary,
+        })
+      } catch {
+        return { installed: '0.0.0', staleness: 'unknown' }
+      }
```

Comment:
IMHO we should split this into 2 try-catch block, so that failing to fetch info from the npm registry won't result in `installed` being `0.0.0`.